src/web/services/search.service.ts

```diff
@@ -120,87 +116,10 @@ export class SearchService {
     return this.httpRequestService.get(ResourceEndpoints.SEARCH_COMMENTS, paramMap);
   }
 
-  getCoursesWithStudents(studentsRes: Students): SearchStudentsListRowTable[] {
-    const { students }: { students: Student[] } = studentsRes;
-
-    const distinctCourses: string[] = Array.from(
-      new Set(students.map((s: Student) => s.courseId)),
+  searchInstructorPrivilege(courseId: string, sectionName: string): Observable<InstructorPrivilege> {
+    return this.instructorService.loadInstructorPrivilege(
```

Comment:
You can just call the method directly. 

- There is no need to use search service as a proxy
- The naming is also a bit misleading as we are not searching instructor privilege.